WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/IssueRefundViewModel.swift

```diff
@@ -235,6 +240,36 @@ extension IssueRefundViewModel {
         let count = state.refundQuantityStore.count()
         return String.pluralize(count, singular: Localization.itemSingular, plural: Localization.itemsPlural)
     }
+
+    /// Return an array of `orderItems` by taking out all previously refunded items
+    ///
+    private static func filterItems(from order: Order, with refunds: [Refund]) -> [OrderItem] {
+        // Flattened array with all items refunded
+        let allRefundedItems = refunds.flatMap { $0.items }
+
+        // Transform `order.items` by substracting the quantity left to refund and evicting those who were fully refunded.
+        return order.items.compactMap { item -> OrderItem? in
+
+            // Calculate how many times an item has been refunded. This number is negative.
+            let timesRefunded = allRefundedItems.reduce(0) { timesRefunded, refundedItem -> Decimal in
```

Comment:
nit: just a thought of optimization - maybe `allRefundedItems` can be reduced to a dictionary that maps from `refundedItem.productOrVariationID` to the number of times this product/variation has been refunded before looping through `order.items`. This way, it saves some time looping through `allRefundedItems` for each order item.

Reply:
I tried that at the beginning but felt the algorithm was a little bit harder to read and understand. Then I weighed that against lopping through the array(which should not have a lot of items) and I decided that it was fast enough 😅